### PR TITLE
Use strict CouchUser.get_by_username in get_active_users_by_email

### PR DESCRIPTION
**Pull request by dannyroberts**
hoping this will solve https://sentry.io/organizations/dimagi/issues/1715324125/?project=136860&query=is%3Aunresolved+password&statsPeriod=14d

### CODE REVIEW

**Review by calellowitz — COMMENTED** [review]
I have no opinion on whether this is a good change anyway, but it does not solve that ticket. More info in the chat, but this does not change anything.

File: corehq/apps/domain/auth.py [hunk 1]
```diff
@@ -225,7 +225,7 @@ def get_active_users_by_email(email):
             yield user
         else:
             # also any mobile workers from TWO_STAGE_USER_PROVISIONING domains should be included
-            couch_user = CouchUser.get_by_username(user.username)
+            couch_user = CouchUser.get_by_username(user.username, strict=True)
             if couch_user.is_commcare_user() and TWO_STAGE_USER_PROVISIONING.enabled(couch_user.domain):
```
Comment (czue) [inline corehq/apps/domain/auth.py:229]:
```suggestion
            if couch_user and couch_user.is_commcare_user() and TWO_STAGE_USER_PROVISIONING.enabled(couch_user.domain):
```

**Review by czue — APPROVED** [review]
this seems like a good change given how relatively rare password resets are. to prevent the 500 I think we also want the check I added, no?

File: corehq/apps/domain/auth.py (hunk 1 above)
Comment (czue) [reply corehq/apps/domain/auth.py:229]: nvm, looks like this was independently added in https://github.com/dimagi/commcare-hq/pull/27758/

**Pull request merged**